test/WebSites/XmlFormattersWebSite/Controllers/HomeController.cs

```diff
@@ -13,7 +14,52 @@ public class HomeController : Controller
         [HttpPost]
         public IActionResult Index([FromBody]DummyClass dummyObject)
         {
+            if (!ModelState.IsValid)
+            {
+                return new ObjectResult(GetModelStateErrorMessages(ModelState))
+                {
+                    StatusCode = 400
+                };
+            }
+
             return Content(dummyObject.SampleInt.ToString());
         }
+
+        // Cannot use 'SerializableError' here as it sanitizes exceptions in model state with generic error message.
+        // Since the tests need to verify the messages, we are doing the following.
+        private List<string> GetModelStateErrorMessages(ModelStateDictionary modelStateDictionary)
```

Comment:
FYI: This code is present in my other PR too. I will remove this and probably move it to a common place and use it.